bytecomp/matching.ml

```diff
@@ -2695,99 +2716,114 @@ let arg_to_var arg cls = match arg with
   The main compilation function.
    Input:
       repr=used for inserting debug events
+      mut= true <=> some field in pattern is mutable and there is are
+        guard or lazy patterns
```

Comment:
Is the "and there are guard or lazy patterns" information really kept in the `mut` variable? In the code above at least it is only about mutability.

Also, the function `compile_match` below does not actually take `mut` argument. Does this comment (and "The 'mut' argument is here..." below) come from an older version of the patch?